docs/Coding/mapgen.rst

```diff
@@ -0,0 +1,569 @@
+  +          +          +-----------+-----------+----------+---------+-------------+--------+
+  |          |          | Dry       | Temperate | Green    | Dry     | Not frozen  | 2      |
+  + Desert   + 40%      +-----------+-----------+----------+---------+-------------+--------+
+  |          |          | Cold      | Dry       | Tropical | Dry     | Not hot     | 1      |
+  +          +          +-----------+-----------+----------+---------+-------------+--------+
+  |          |          | Frozen    | Dry       | \-       | Dry     | Frozen      | 1      |
+  +----------+----------+-----------+-----------+----------+---------+-------------+--------+
+  |          |          | Mountain  | Green     | \-       | All     | All         | 2      |
+  + Mountain + 20%      +-----------+-----------+----------+---------+-------------+--------+
+  |          |          | Mountain  | \-        | Green    | All     | All         | 1      |
+  +----------+----------+-----------+-----------+----------+---------+-------------+--------+
+  |          |          | Wet       | Tropical  | Foliage  | Not dry | Tropical    | 1      |
+  +          +          +-----------+-----------+----------+---------+-------------+--------+
+  | Swamp    | 80%      | Wet       | Temperate | Foliage  | Not dry | Hot         | 2      |
+  +          +          +-----------+-----------+----------+---------+-------------+--------+
+  |          |          | Wet       | Cold      | Foliage  | Not dry | Not hot     | 1      |
+  +----------+----------+-----------+-----------+----------+---------+-------------+--------+
+
+.. _mapgen-placement:
+
+Player Placement
+----------------
+
+.. table:: Mode chosen by the generator to generate start positions
+  :widths: auto
+  :align: right
+
+  ============ =======
+  Generator    Default
+  ============ =======
+  ``FAIR``     Placed by the generator
+  ``FRACTURE`` ``ALL``
+  ``FRACTAL``  ``ALL``
+  ``ISLAND``   Always ``SINGLE``
+  ``RANDOM``   ``2or3``
+  Scenarios    ``ALL``
+  ============ =======
+
+The final step in the map generator is to to place players on the map. The
+method used to do so is set by the user, but it can also can depend on the
+generator if ``startpos`` is set to ``DEFAULT``, as listed on the right.
+Starting positions are allocated using the chosen method. If a method fails,
+another method is tried in the following order: ``SINGLE``, ``2or3``, ``ALL``,
+and finally ``VARIABLE``.
+
+Placement tries to find fair starting positions using a "tile value" metric,
+computed as the sum of all outputs produced by the tile (food, production, and
+trade). If the initial workers can build a road, irrigation, or mine, half of
+the best possible bonus is counted towards the value of the tile, rounded down.
+Specials are taken into account but no government bonus is applied. This gives
+the following values for common terrains:
+
+.. table:: Tile value for the most common terrains
```

Comment:
s/terrains/terrain

Reply:
This is also about terrain "types" from the ruleset